server/src/test/java/org/elasticsearch/common/bytes/ZeroBytesReference.java

```diff
@@ -56,20 +58,23 @@ public BytesRef toBytesRef() {
 
     @Override
     public BytesRefIterator iterator() {
-        if (length <= PageCacheRecycler.BYTE_PAGE_SIZE) {
-            return super.iterator();
-        }
-        final byte[] buffer = new byte[PageCacheRecycler.BYTE_PAGE_SIZE];
+        final byte[] buffer = new byte[Math.min(length, PageCacheRecycler.BYTE_PAGE_SIZE)];
         return new BytesRefIterator() {
-
             int remaining = length;
+            BytesRef current;
 
             @Override
             public BytesRef next() {
+                if (current != null) {
```

Comment:
Not sure we need to track `current` here, we can just look at `buffer` directly.

Reply:
doh! of course....